adb/sign_m2crypto.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import hashlib
+
+from M2Crypto import RSA
+
+import adb_protocol
+
+
+class M2CryptoSigner(adb_protocol.AuthSigner):
+  """AuthSigner using M2Crypto."""
+
+  def __init__(self, rsa_key_path):
```

Comment:
Actually, on second thought, make M2CryptoSigner a function that returns an instance of a slightly different class that takes in the public key as a string and the RSA object in its constructor:

```py
def M2CryptoSigner(rsa_key_path):
  with open(rsa_key_path + '.pub') as rsa_pub_file:
    public_key = rsa_pub_file.read()
  rsa_key = RSA.load_key(rsa_key_path)
  return M2CryptoAuthSigner(public_keys, rsa_key)

class M2CryptoAuthSigner(adb_protocol.AuthSigner):
  def __init__(self, public_key, rsa_key):
    self.public_key = public_ key
    self.rsa_key = rsa_key

  ...
```